source/community.applications/usr/local/emhttp/plugins/community.applications/include/helpers.php

```
@@ -188,76 +276,166 @@ function ca_file_put_contents($filename,$data,$flags=0) {
+  if ( $download_total > 0 ) {
+    $percentage = intval($download_current / $download_total * 100);
+    ca_publish("ca_downloadProgress",basename($testProgress['url'])." - ".MakeReadable($download_current)." of ".MakeReadable($download_total)." at ".MakeReadable($testProgress['speed_download'])."/s ($percentage%)");
+  } else {
+    ca_publish("ca_downloadProgress",basename($testProgress['url'])." - ".MakeReadable($download_current)." at ".MakeReadable($testProgress['speed_download'])."/s");
+  }
 }
 
-function download_json($url,$path="",$bg=false,$timeout=45) {
+function ca_publish($endpoint,$message) {
+  if ( ! function_exists("publish_noDupe") ) {
+    return publish($endpoint,$message);
```

PHPStan (2.1.51)

[error] Result of function publish (void) is used.


(function.void)